Add NuGet 6.12 release notes (#3349)
NuGet/docs.microsoft.com-nuget: docs/release-notes/NuGet-6.12.md

<sup>1</sup> Installed with Visual Studio 2022 with any .NET workload

## Summary: What's New in 6.11.1

NuGet 6.12.1 is available in Visual Studio 17.12.0 and the.NET 9.0.101 SDK.

### Issues fixed in this release

* Deserializing an empty version range in a package dependency fails in .NET SDK 9.0.100-rc.2 [#13869](https://github.com/NuGet/Home/issues/13869)

> ## Summary: What's New in 6.12

Reviewer: 6.12 has an unlikely bug in the new resolver, https://github.com/NuGet/Home/issues/13890. 

Should we include it in these docs?
How should we include it there?

Author: I added a new section for known issues

NuGet 6.12.0 is available in the .NET 9.0.100 SDK.

> * Add property for toggling the to the previous NuGet resolver: RestoreUseLegacyDependencyResolver - [#13700](https://github.com/NuGet/Home/issues/13700)

Reviewer: If we have the new dependency resolver chnage near the top here, do we still need this?

* Owner profile hyperlinks needed in Details Pane of PM UI - [#13686](https://github.com/NuGet/Home/issues/13686)

* Bubble-up Known Vulnerability Indicators in Solution Explorer for Transitive Packages - [#13636](https://github.com/NuGet/Home/issues/13636)

> * Map branch name from sourcelink to RepositoryBranch for NuGet pack - [#13625](https://github.com/NuGet/Home/issues/13625)

Reviewer: This probably can be moved to changes, not sure if it's worth calling out in the big features.

* Enable `dotnet nuget why` on non-SDK style projects - [#13576](https://github.com/NuGet/Home/issues/13576)

* NuGetAuditSuppress for packages.config - [#13575](https://github.com/NuGet/Home/issues/13575)

* [Feature] Enable Transitive Dependencies for Solution-level in Visual Studio - [#13216](https://github.com/NuGet/Home/issues/13216)

* Audit security vulnerabilities without adding nuget.org as package source - [#12698](https://github.com/NuGet/Home/issues/12698)

* Roll-out new breaking change process for SDK tools, respect SdkAnalysisLevel - [#13309](https://github.com/NuGet/Home/issues/13309)

### Breaking changes

* NuGetAudit should check transitive packages by default when the .NET 9 SDK is installed [#13293](https://github.com/NuGet/Home/issues/13293)

* Change NuGetAuditMode default to all for non-SDK style projects - [#13584](https://github.com/NuGet/Home/issues/13584)

> * [Epic]Deprecate http usage: Promote from warning to error - [#13289](https://github.com/NuGet/Home/issues/13289)

Reviewer: ```suggestion
* Deprecate http usage: Promote from warning to error - [#13289](https://github.com/NuGet/Home/issues/13289)
```

Should we just move this to the What's New?

I think it's really up to us where we want to include it.

> ### Issues fixed in this release

Reviewer 2: Thanks for identifying that the release notes are missing a customer facing change. Sorry for the delay in creating the issue. https://github.com/NuGet/Home/issues/13891

* Reduce allocations in TokenSegment.TryMatch - [#12728](https://github.com/NuGet/Home/issues/12728)

* Use `SDKAnalysisLevel` in restore "https everywhere: promote from warning to error" - [#13546](https://github.com/NuGet/Home/issues/13546)

* tweak wording of NU1603 - [#13446](https://github.com/NuGet/Home/issues/13446)

* Default Package icon shown even when embedded icon file exists on disk - [#13766](https://github.com/NuGet/Home/issues/13766)

* Navigation telemetry for hyperlinks: License, ReportAbuse, Readme, ProjectUrl - [#13749](https://github.com/NuGet/Home/issues/13749)

* Navigation telemetry for Owner Profile URLs in PM UI - [#13738](https://github.com/NuGet/Home/issues/13738)

> * NuGet's dependency graph resolution algorithm does not scale well for large graphs - [#13692](https://github.com/NuGet/Home/issues/13692)

Reviewer: This should probably in the What's new as one of the 2-3 flagship changes.

<sup>1</sup> Installed with Visual Studio 2022 with any .NET workload

## Known Issues

* Project and package in the same graph with the same name but different dependencies may lead to incorrect versions of the dependencies of that id [#13888](https://github.com/NuGet/Home/issues/13888)
* VS PM UI shows warning icon about package vulnerability even after upgrade [#13866](https://github.com/NuGet/Home/issues/13866)
* dotnet nuget why reports missing argument, even though it ran [#13908](https://github.com/NuGet/Home/issues/13908)

## Summary: What's New in 6.12.1

NuGet 6.12.1 is available in Visual Studio 17.12.0 and the .NET 9.0.101 SDK.

### Issues fixed in this release

* Deserializing an empty version range in a package dependency fails in .NET SDK 9.0.100-rc.2 [#13869](https://github.com/NuGet/Home/issues/13869)

## Summary: What's New in 6.12

NuGet 6.12.0 is available in the .NET 9.0.100 SDK.

> * NuGet's dependency graph resolution algorithm does not scale well for large graphs - [#13692](https://github.com/NuGet/Home/issues/13692)

Reviewer: Not sure if I commented on this originally, but It's probably ok to take creative freedom and write it in a way that's nicer. 

```suggestion
* Introduce optimized dependency resolver that scales well for large graphs - [#13692](https://github.com/NuGet/Home/issues/13692)
* ```

Author: > take creative freedom

ok!

```suggestion
* The hypotenuse of the monad is tangential to the flux capacitor second derivitive  - [#13692](https://github.com/NuGet/Home/issues/13692)
```
